### DISCUSSION TITLE
Configuring REPL colours

### DISCUSSION THREAD

**Discussion by dimitri-b** [erezsh/Preql#29] [Q&A] [answered]
For some reason in my terminal certain keywords (e.g. `func`) appear in a dark blue colour that fades into the dark background and is almost invisible. Switching to a light scheme has the same problem - these keywords blend into the background.
I was wondering if there is a way to change that colour via config and make it stick between sessions?

**Comment by erezsh** [comment, marked as answer]
Sorry that the colors aren't optimal! There is no config option to change them, at the moment. But I agree it would be nice to have.

As an immediate solution, I would suggest editing `preql/repl.py`. You will see a class there:

```python
class PreqlStyle(Style):
    default_style = ""
    styles = {
        Generic:                'ansigray',
        Comment:                'italic #888',
        Keyword:                'bold #00f',
        Name:                   '#fff',
        Name.Function:          'bold #8f8',
        Name.Class:             'bold #0f0',
        String:                 'ansigreen',
        Number:                 'ansicyan',
        Operator:               'ansigray',
        Error:                  'bg:ansired ansigray',
    }
```

You can pick any color you want.

Specifically the relevant line there is:

```python
        Keyword:                'bold #00f',
```

Unfortunately it will reset next time you update Preql, but I will try to include a way to config the colors by then.

**Reply by erezsh** [reply]
Update: I now pushed to `master` basic support for a config file.

(being read from `~/.preql_conf.json`, or specified using the `-c` switch)

**Reply by dimitri-b** [reply]
Does it mean I can install from GitHub and add the `~/.preql_conf.json` file to fix this? If yes - what would be the format / syntax - same or similar to the class in your example?

**Reply by erezsh** [reply]
Yes. Here's an example file:

```json
{
	"color_scheme": {
		"string": "#ffcccc",
		"keyword": "#ccccff"
	}
}
```

**Reply by dimitri-b** [reply]
I tried to change colour via `.preql_conf.json` to a shade of purple, for a test - it didn't work, nothing changed.

Here is what it looks like in Pycharm and OS terminals, for some reason OS terminal doesn't have this issue.

<img width="500" alt="2021-08-19_22-21-58" src="https://user-images.githubusercontent.com/4684795/130068249-976792e5-13eb-458c-8209-97c62c310eb3.png">

**Reply by erezsh** [reply]
Did you put it in your home directory? Maybe try using the `-c` switch to make sure it's being read?

Yes, the colors are readable in Windows too. It's a little hard to get them to be consistent between different platforms. But I had hoped the latest commit would help.

**Reply by dimitri-b** [reply]
Yes, it's in my home dir.

No luck with `-c` switch either - am I doing somethnig wrong?

<img width="664" alt="2021-08-20_07-17-17" src="https://user-images.githubusercontent.com/4684795/130145804-e3d51e6d-778b-407b-aeb0-1af2e2c27cdb.png">

**Reply by erezsh** [reply]
As I suspected, you haven't been using the latest commit. It's probably due to pathing or a similar collision.

Anyway, I just published version `0.2.4`. Run `pip install -U preql` and try it again.

P.S. I also plan to add schema support soon, probably by next week.

**Reply by dimitri-b** [reply]
It worked this time, thank you.
